### PR TITLE
[processor/transform] Update metrics to use tqlmetrics

### PR DESCRIPTION
**Pull request by TylerHelmuth**
**Description:**
Updates metrics to use the tqlmetrics context from pkg/telemetryquerylanguage. No functional change to the transform processor.

**Link to tracking Issue:**
#11751 

**Testing:** 
Ran unit tests

### CODE REVIEW

File: processor/transformprocessor/internal/metrics/processor.go [hunk 1]
```diff
@@ -41,64 +43,64 @@ func NewProcessor(statements []string, functions map[string]interface{}, setting
...
+func (p *Processor) handleExponetialHistogramDataPoints(ctx tqlmetrics.MetricTransformContext, dps pmetric.ExponentialHistogramDataPointSlice) {
 	for i := 0; i < dps.Len(); i++ {
-		ctx.dataPoint = dps.At(i)
+		ctx.DataPoint = dps.At(i)
 		p.callFunctions(ctx)
 	}
 }
 
-func (p *Processor) handleSummaryDataPoints(ctx metricTransformContext, dps pmetric.SummaryDataPointSlice) {
+func (p *Processor) handleSummaryDataPoints(ctx tqlmetrics.MetricTransformContext, dps pmetric.SummaryDataPointSlice) {
```
Comment (bogdandrutu) [inline processor/transformprocessor/internal/metrics/processor.go:96]: `tqlmetrics.MetricTransformContext` can be renamed to `tqlmetrics.TransformContext`

**Review by bogdandrutu — APPROVED** [review]

File: processor/transformprocessor/internal/metrics/processor.go (hunk 1 above)
Comment (bogdandrutu) [reply processor/transformprocessor/internal/metrics/processor.go:96]: Separate PR, we can look into these kind of name improvements.

Comment (TylerHelmuth) [reply processor/transformprocessor/internal/metrics/processor.go:96]: Should probably be consistent and do that style of naming for all 3 contexts.   I'll do that tomorrow.

Comment (TylerHelmuth) [reply processor/transformprocessor/internal/metrics/processor.go:96]: @bogdandrutu any issue with the context's structs names exactly matching the name of the [tql interface](TransformContext) they are meeting, and also each other?  I know it is all different packages, but might be confusing

Comment (TylerHelmuth) [reply processor/transformprocessor/internal/metrics/processor.go:96]: Here is what it looks like: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/12895

**Pull request merged**